nemo/collections/asr/modules/audio_modules.py

```diff
@@ -718,6 +718,257 @@ def forward(self, input: torch.Tensor, input_length: torch.Tensor) -> Tuple[torc
         return masks, mask_length
 
 
+class MaskEstimatorGSS(NeuralModule):
+    """Estimate masks using guided source separation with a complex
+    angular Central Gaussian Mixture Model (cACGMM).
+
+    Notation is approximately following [1], where `gamma` denotes
+    the time-frequency mask, `alpha` denotes the mixture weights,
+    and `BM` denotes the shape matrix. Additionally, the provided
+    source activity is denoted as `activity`.
+
+    Args:
+        num_iterations: Number of iterations for the EM algorithm
+        eps: Small value for regularization
+        dtype: Data type for internal computations (default `torch.cdouble`)
+
+    References:
+        [1] Ito et al., Complex Angular Central Gaussian Mixture Model for Directional Statistics in Mask-Based Microphone Array Signal Processing, 2016
+        [2] Boeddeker et al., Front-End Processing for the CHiME-5 Dinner Party Scenario, 2018
```

Comment:
For those who are coming from CHiME-7/8, could you add a short description about which part of this NeMo implementation is following Boeddeker's 2018 paper?